daffodil-test/src/test/scala/org/apache/daffodil/runtime1/layers/IPv4ChecksumLayer.scala

```diff
@@ -18,48 +18,77 @@
 package org.apache.daffodil.runtime1.layers
 
 import java.nio.ByteBuffer
+import java.util.Optional
+import scala.collection.JavaConverters._
 
 import org.apache.daffodil.lib.exceptions.Assert
-import org.apache.daffodil.runtime1.processors.ParseOrUnparseState
-import org.apache.daffodil.runtime1.processors.VariableRuntimeData
+import org.apache.daffodil.runtime1.api.DFDLPrimType
+import org.apache.daffodil.runtime1.layers.api.JLayerLengthKind
+import org.apache.daffodil.runtime1.layers.api.Layer
+import org.apache.daffodil.runtime1.layers.api.LayerChecksumMixin
+import org.apache.daffodil.runtime1.layers.api.LayerRuntime
+import org.apache.daffodil.runtime1.layers.api.LayerVariables
 
 import passera.unsigned.UShort
 
+object IPv4ChecksumLayer {
+  val name: String = "IPv4Checksum"
+  val vars: LayerVariables =
+    LayerVariables(
+      prefix = "chksum",
+      namespace = "urn:org.apache.daffodil.layers.IPv4Checksum",
+      variables = Seq(
+        (name, DFDLPrimType.UnsignedShort),
+      ).asJava,
+    )
+}
+
 /**
  *  The layer transform computes the checksum of the header data.
  *  per IETF RFC 791.
  *
- *  The data has a well-known fixed length, so layerLengthKind is always 'implicit'.
+ *  The data has a well-known fixed length, so layerLengthKind is 'implicit' or not specified.
  */
-final class IPv4Checksum(
-  name: String,
-  layerRuntimeInfo: LayerRuntimeInfo,
-  outputVar: VariableRuntimeData,
-) extends ByteBufferExplicitLengthLayerTransform[Int](
-    layerRuntimeInfo,
-    name,
-    inputVars = Seq(),
-    outputVar: VariableRuntimeData,
-  ) {
+final class IPv4ChecksumLayer()
+  extends Layer(
+    IPv4ChecksumLayer.name,
+    supportedLayerLengthKinds =
+      Seq(JLayerLengthKind.Implicit).asJava, // Implicit implies you cannot specify layerLength.
+    supportedLayerLengthUnits = Seq().asJava,
+    isRequiredLayerEncoding = false,
+    optLayerVariables = Optional.of(IPv4ChecksumLayer.vars),
+  )
+  with LayerChecksumMixin // Checksums are easily expressed as a compute method on a byte buffer.
+  {
 
   /**
    * This layer is always exactly 20 bytes long.
    */
-  override protected def layerBuiltInConstantLength = Some(20L)
+  private def lenInBytes = 20 // bytes
+
+  override def optLayerBuiltInConstantLength: Optional[Int] = Optional.of(lenInBytes)
```

Comment:
Could just punt on this built in fixed length stuff and just use layerLengthKind explicit for these. 

Or, could just use layerLengthKind 'implicit' and implement 'explicit' using an element of specified length that surrounds the part that needs to have the fixed length.